packages/labs/react/src/create-component.ts

```diff
@@ -315,7 +317,11 @@ export function createComponent<
         if (
           eventProps.has(k) ||
           (!reservedReactProperties.has(k) &&
-            !(k in HTMLElement.prototype) &&
+            // NODE_MODE check below is to support React SSR where HTMLElement
+            // might not be defined
+            !(NODE_MODE && typeof HTMLElement === 'undefined'
```

Comment:
in https://github.com/lit/lit/pull/3410 there is no base prototypes being used in this space, only the provided element.
```
        if (eventProps.has(k) || k in element.prototype) {
          this._elementProps[k] = v;
          continue;
        }
```

This check needs to be done at the `setProperty` function. And that's where the `HTMLElement` or whatever prototype if-ing would be great

also nit but `if (NODE_MODE` is always false than unnecessary?

I would advise rebasing this off of https://github.com/lit/lit/pull/3410

Reply:
This PR aims to simply add the node build without changing any of the client-side runtime behavior from the current state of the package. Also I believe you mean https://github.com/lit/lit/pull/3128. I've added where I'd move the `NODE_BUILD` check for that branch here https://github.com/lit/lit/pull/3410#issuecomment-1297853312

I don't particularly see a reason why one has to go before the other but I'm okay to wait on this if that's what we'd like.

Reply:
After sync, I understand there will mostly likely need to be some kind of node check. Marking as resolved! :)